JASP-Engine/JASP/R/ttestindependentsamples.R

```diff
@@ -196,294 +199,295 @@ TTestIndependentSamples <- function(jaspResults, dataset = NULL, options, ...) {
   
   if (options$effectSizesType == "cohensD")
     effSize <- "cohen"
   else if (options$effectSizesType == "glassD")
     effSize <- "glass"
   else if (options$effectSizesType == "hedgesG")
     effSize <- "hedges"
+
+  levels <- levels(dataset[[ .v(options$groupingVariable) ]])
   
-  nameOfEffectSizeParametric <- switch(effSize, 
-                                       cohen  = gettext("Cohen's d"), 
-                                       glass  = gettext("Glass' delta"),
-                                       hedges = gettext("Hedges' g"))
-  
-  variables  <- options$variables
-  if (length(variables) == 0) 
-    variables <- "."
+  if (options$hypothesis == "groupOneGreater" || options$hypothesis == "groupTwoGreater") {
+    directionNote <- ifelse(options$hypothesis == "groupOneGreater", gettext("greater"), gettext("less"))
+    table$addFootnote(gettextf("For all tests, the alternative hypothesis specifies that group %1$s is %2$s than group %3$s.",
+                                                paste("<em>", levels[1], "</em>"), directionNote, paste("<em>", levels[2], "</em>")))
+  }
   
-  if(!ready) {
-    for (variable in variables) 
-      jaspResults[["ttest"]]$addRows(list(v = variable))
-  } else {
-    levels <- levels(dataset[[ .v(options$groupingVariable) ]])
+  ## for each variable specified, run each test that the user wants
+  for (variable in options$variables) {
+    errors <- .hasErrors(dataset, 
+                         message = 'short', 
+                         type = c('observations', 'variance', 'infinity'),
+                         all.target = variable, 
+                         all.grouping = options$groupingVariable,
+                         observations.amount = '< 2')
     
-    if (options$hypothesis == "groupOneGreater" || options$hypothesis == "groupTwoGreater") {
-      directionNote <- ifelse(options$hypothesis == "groupOneGreater", gettext("greater"), gettext("less"))
-      jaspResults[["ttest"]]$addFootnote(gettextf("For all tests, the alternative hypothesis specifies that group %1$s is %2$s than group %3$s.",
-                                                  paste("<em>", levels[1], "</em>"), directionNote, paste("<em>", levels[2], "</em>")))
-    }
-    
-    groupingData <- dataset[[ .v(options$groupingVariable) ]]
-    ## for each variable specified, run each test that the user wants
-    for (variable in options$variables) {
-      errors <- .hasErrors(dataset, 
-                           message = 'short', 
-                           type = c('observations', 'variance', 'infinity'),
-                           all.target = variable, 
-                           all.grouping = options$groupingVariable,
-                           observations.amount = '< 2')
-      variableData <- dataset[[ .v(variable) ]]
+    ## test is a number, indicating which tests should be run
+    for (test in seq_len(length(optionsList$whichTests))) {
```

Comment:
If you do not want to fix this indication by number now, then I would at least change this
```suggestion
    for (test in seq_along(optionsList$whichTests)) {
```